shared_model/backend/protobuf/commands/proto_create_role.hpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "interfaces/commands/create_role.hpp"
+
+#ifndef IROHA_PROTO_CREATE_ROLE_HPP
+#define IROHA_PROTO_CREATE_ROLE_HPP
+
+namespace shared_model {
+  namespace proto {
+    class CreateRole final : public CopyableProto<interface::CreateRole,
+                                                  iroha::protocol::Command,
+                                                  CreateRole> {
+     public:
+      template <typename CommandType>
+      explicit CreateRole(CommandType &&command)
+          : CopyableProto(std::forward<CommandType>(command)),
+            create_role_(detail::makeReferenceGenerator(
+                proto_, &iroha::protocol::Command::create_role)),
+            role_permissions_([this] {
+              std::set<std::string> perms;
```

Comment:
This can be simplified with the following:
```C++
return boost::accumulate(
    create_role_->permissions(),
    PermissionsType{},
    [](auto &&acc, const auto &perm) {
      acc.insert(iroha::protocol::RolePermission_Name(
          static_cast<iroha::protocol::RolePermission>(perm)));
      return std::forward<decltype(acc)>(acc);
    });
```